rosdoc2/verbs/build/builders/doxygen_builder.py

```diff
@@ -77,6 +77,15 @@ def __init__(self, builder_name, builder_entry_dictionary, build_context):
 
         assert self.builder_type == 'doxygen'
 
+        # If the build type is not `ament_cmake/cmake`, there is no reason
+        # to create a doxygen builder.
+        if (self.build_context.build_type not in ('ament_cmake', 'cmake') and
+                not self.build_context.always_run_doxygen):
+            logger.info(
+                f"The package build type is not 'ament_cmake' or 'cmake', hence the"
+                f"'{self.builder_type}' builder was not configured")
+            return None
```

Comment:
nitpick: I prefer this style of wrapping long conditionals so you never indent more than once between two consecutive lines but also avoid indentation ambiguity:

```suggestion
        if (
            self.build_context.build_type not in ('ament_cmake', 'cmake') and
            not self.build_context.always_run_doxygen
        ):
            logger.info(
                f"The package build type is not 'ament_cmake' or 'cmake', hence the"
                f"'{self.builder_type}' builder was not configured")
            return None
```